homeassistant/components/ring/switch.py

```diff
@@ -59,20 +66,22 @@ class SirenSwitch(BaseRingSwitch):
     _attr_translation_key = "siren"
     _attr_icon = SIREN_ICON
 
-    def __init__(self, config_entry_id, device):
+    def __init__(self, device: RingGeneric, coordinator: RingDataCoordinator) -> None:
         """Initialize the switch for a device with a siren."""
-        super().__init__(config_entry_id, device, "siren")
+        super().__init__(device, coordinator, "siren")
         self._no_updates_until = dt_util.utcnow()
         self._attr_is_on = device.siren > 0
 
     @callback
-    def _update_callback(self):
+    def _handle_coordinator_update(self):
         """Call update method."""
         if self._no_updates_until > dt_util.utcnow():
             return
 
-        self._attr_is_on = self._device.siren > 0
-        self.async_write_ha_state()
+        device: Optional[RingGeneric]
+        if (device := self._get_coordinator_device()) and hasattr(device, "siren"):
```

Comment:
Can we use here maybe an instance check?

Reply:
Replaced with an instance check and also in the `light.py` where there was a similar check.